### ISSUE TITLE
Does an observation refer to a time instance or a time interval?

### ISSUE THREAD

**Issue by krlmlr** [christophsax/tsbox#29]
Likewise: Is this just a pointwise measurement or an aggregation over a time interval?

How is this information stored in time series object? How is it preserved when converting back and forth between time series objects and tibbles?

CC @edzer @etiennebr.

**Comment by christophsax** [comment]
We don't know. A time stamp may point to the *start* of an interval or to a point in time. 

E.g., share prices - point in time. GDP - start of an interval.

Since this is not known to data frames or ts objects, no need to preserve anything.

I guess we usually don't care except when converting between frequencies. If we are converting between frequencies, we need to specify how to do that (mean, sum, first or last value).

**Comment by christophsax** [comment]
Here's an example for frequency conversion:
``` r
library(tsbox)
ts_frequency(ts_tbl(AirPassengers), to = "quarter", fun = sum)
#> Loading required namespace: tibble
#> # A tibble: 48 x 2
#>    time       value
#>    <date>     <dbl>
#>  1 1949-01-01  362.
#>  2 1949-04-01  385.
#>  3 1949-07-01  432.
#>  4 1949-10-01  341.
#>  5 1950-01-01  382.
#>  6 1950-04-01  409.
#>  7 1950-07-01  498.
#>  8 1950-10-01  387.
#>  9 1951-01-01  473.
#> 10 1951-04-01  513.
#> # ... with 38 more rows
ts_frequency(ts_tbl(EuStockMarkets), to = "quarter", fun = mean)
#> # A tibble: 116 x 3
#>    id    time                value
#>    <chr> <dttm>              <dbl>
#>  1 DAX   1991-07-01 02:00:00 1627.
#>  2 DAX   1991-10-01 01:00:00 1577.
#>  3 DAX   1992-01-01 01:00:00 1690.
#>  4 DAX   1992-04-01 02:00:00 1758.
#>  5 DAX   1992-07-01 02:00:00 1603.
#>  6 DAX   1992-10-01 01:00:00 1501.
#>  7 DAX   1993-01-01 01:00:00 1627.
#>  8 DAX   1993-04-01 02:00:00 1653.
#>  9 DAX   1993-07-01 02:00:00 1856.
#> 10 DAX   1993-10-01 01:00:00 2068.
#> # ... with 106 more rows
```

**Comment by edzer** [comment]
So, so something returning from `ts_frequency` could know, but then doesn't. Shouldn't it also matter for plotting, where you place values: at the start of the year, or halfway the year, for yearly data?

**Comment by christophsax** [comment]
Yes, that's a good point, it should also matter for plotting. But I guess there is a long history of ignoring it, and this is also what R base plots do:

``` r
ts.plot(EuStockMarkets)
ts.plot(AirPassengers)
```

Also, for annual values, I don't see anything wrong with these graphs, even when I use the `last` aggregation function (which may make sense for stock indices). It's just that we interpret the tick as belonging to the whole year instead of belonging to January 1:

``` r
library(tsbox)
ts_plot(
  ts_frequency(EuStockMarkets, to = "year", fun = data.table::last), 
  title = "EuStockMarkets", subtitle = "End of Year Value"
)
```

![](https://i.imgur.com/2G8JA3c.png)

``` r

ts_plot(
  ts_frequency(AirPassengers, to = "year", fun = sum), 
  title = "AirPassengers", subtitle = "Total Annual Value"
)
```

![](https://i.imgur.com/MGAJ9Q9.png)

For quarterly, values, things are more complicated. The Stock Market series if off by one quarter, as the end of period value is plotted at the beginning of the period.

``` r
ts_plot(
  ts_frequency(EuStockMarkets, to = "quarter", fun = data.table::last), 
  title = "EuStockMarkets", subtitle = "End of Quarter Value"
)
```

![](https://i.imgur.com/JnOWAAO.png)

where a better graph would be probably:

``` r
ts_plot(
  ts_lag(ts_frequency(EuStockMarkets, to = "quarter", fun = data.table::last), -1), 
  title = "EuStockMarkets", subtitle = "End of Quarter Value"
)
```

![](https://i.imgur.com/OJ1m8Qv.png)

**To summarize**: If we knew if a series is aggregated by `sum`, `mean`, `first` or `last`, that wold be helpful for plotting and aggregation. But then for most of the time series operations we still don't care, so I don't think it makes sense to carry this around.
